src/config.rs

```diff
@@ -163,49 +174,131 @@ impl Config {
+				args.homedir = Some(String::from(DEFAULT_HOMEDIR));
+				args.outdir = Some(String::from(DEFAULT_OUTDIR));
+			}
+		}
+		match self.general.as_ref() {
+			Some(general) => {
+				args.splash = general.splash.unwrap_or_default();
+				args.tick_rate = general.tick_rate.unwrap_or(DEFAULT_TICK_RATE);
+				args.color = general
+					.color
+					.as_ref()
+					.map(|color| Color::from(color.as_ref()))
+					.unwrap_or(default_color);
+				args.style = general
+					.style
+					.as_ref()
+					.map(|style| {
+						Style::from_str(style.as_ref(), true)
+							.unwrap_or_default()
+					})
+					.unwrap_or_default();
+				args.file_explorer = general
+					.file_explorer
+					.as_ref()
+					.cloned()
+					.unwrap_or(default_file_explorer);
+				args.detail_level = general.detail_level.unwrap_or(
+					KeyDetail::from_str(DEFAULT_DETAIL_LEVEL, true)
+						.unwrap_or_default(),
+				);
+			}
+			None => {
+				args.splash = DEFAULT_SPLASH;
+				args.tick_rate = DEFAULT_TICK_RATE;
+				args.color = default_color;
+				args.style = default_style;
+				args.file_explorer = default_file_explorer;
+			}
 		}
-		args.default_key = self.gpg.default_key.clone();
-		args.tick_rate = self.general.tick_rate;
-		args.color = Color::from(self.general.color.as_ref());
-		args.style =
-			Style::from_str(&self.general.style, true).unwrap_or_default();
-		args.detail_level = self.general.detail_level.unwrap_or_default();
-		args.file_explorer = self.general.file_explorer.clone();
 		args
 	}
 }
-
 #[cfg(test)]
 mod tests {
 	use super::*;
 	use pretty_assertions::assert_eq;
+	use std::fs::{self, File};
+	use std::io::Write;
 	use std::path::PathBuf;
+
 	#[test]
 	fn test_parse_config() -> Result<()> {
-		let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+		let path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
 			.join("config")
 			.join(format!("{}.toml", env!("CARGO_PKG_NAME")))
 			.to_string_lossy()
 			.into_owned();
-		if let Some(global_path) = Config::get_default_location() {
-			path = global_path;
-		}
 		let mut config = Config::parse_config(&path)?;
-		config.gpg.default_key = Some(String::from("test_key"));
+		if let Some(ref mut gpg) = config.gpg {
+			gpg.default_key = Some(String::from("test_key"));
+		}
 		let args = config.update_args(Args::default());
 		assert_eq!(Some(String::from("test_key")), args.default_key);
 		Ok(())
 	}
 
+	#[test]
+	fn test_args_partial_config_general() -> Result<()> {
+		let mut temp_file = File::create("config/temp.toml")?;
+		temp_file.write_all("[general]\n   splash = true\n".as_bytes())?;
+		let tmp_path = PathBuf::from("config/temp.toml");
+		if let Ok(config) = Config::parse_config(&tmp_path.to_string_lossy()) {
+			let args = config.update_args(Args::default());
+			// [general]
+			assert_eq!(args.splash, true); // supplied
+			assert_eq!(args.tick_rate, 250_u64);
+			// [gpg]
+			assert_eq!(args.armor, false);
+			assert_eq!(args.default_key, None);
+		}
+		temp_file.flush()?;
```

Comment:
What's the benefit of using `flush` here? Is it really needed?

Reply:
Ah, you are correct as I forgot to remove that when I changed from write to write_all